packages/@webex/plugin-meetings/src/breakouts/index.ts

```diff
@@ -0,0 +1,206 @@
+ */
+const Breakouts = WebexPlugin.extend({
+  namespace: MEETINGS,
+
+  collections: {
+    breakouts: BreakoutCollection
+  },
+
+  props: {
+    allowBackToMain: 'boolean', // only present when in a breakout session
+    delayCloseTime: 'number', // appears once breakouts start
+    enableBreakoutSession: 'boolean', // appears from the moment you enable breakouts
+    groupId: 'string', // appears from the moment you enable breakouts
+    name: 'string', // only present when in a breakout session
+    sessionId: 'string', // appears from the moment you enable breakouts
+    sessionType: 'string', // appears from the moment you enable breakouts
+    startTime: 'string', // appears once breakouts start
+    status: 'string', // only present when in a breakout session
+    url: 'string', // appears from the moment you enable breakouts
+    locusUrl: 'string', // the current locus url
+  },
+
+  children: {
+    currentBreakoutSession: Breakout,
+  },
+
+  derived: {
+    isInMainSession: {
+      deps: ['sessionType'],
+      /**
+       * Returns true if the user is in the main session
+       * @returns {boolean}
+       */
+      fn() {
+        return this.sessionType === BREAKOUTS.SESSION_TYPES.MAIN;
+      }
+    },
+  },
+
+  /**
+   * initialize for the breakouts
+   * @returns {void}
+   */
+  initialize() {
+    this.listenTo(this, 'change:status', () => {
+      if (this.status === BREAKOUTS.STATUS.CLOSING) {
+        this.trigger(BREAKOUTS.EVENTS.BREAKOUTS_CLOSING);
+      }
+    });
+    this.debouncedQueryRosters = debounce(this.queryRosters, 10, {
+      leading: true,
+      trailing: false
+    })
+    this.listenTo(this.breakouts, 'add', () => {
+      this.debouncedQueryRosters();
+    });
+    this.listenToBroadcastMessages();
+    this.listenToBreakoutRosters();
+  },
+
+  /**
+   * Calls this to clean up listeners
+   */
+  cleanUp() {
+    this.stopListening()
+  },
+
+  /**
+   * Update the current locus url of the meeting
+   */
+  locusUrlUpdate(locusUrl) {
+    this.set('locusUrl', locusUrl);
+  },
+
+  /**
+   * The initial roster lists need to be quried because you don't
+   * get a breakout.roster event when you join the meeting
+   */
+  queryRosters() {
+    this.webex.request(
+      {uri: `${this.url}/roster`, qs: {locusUrl: btoa(this.locusUrl)}
+    }).then((result) => {
+      const {body: {rosters}} = result;
+      
+      rosters.forEach(({locus}) => {
+        this.handleRosterUpdate(locus);
+      })
+
+      this.trigger(BREAKOUTS.EVENTS.MEMBERS_UPDATE);
+    })
+  },
+
+  handleRosterUpdate(locus) {
+    const sessionId = locus.controls?.breakout?.sessionId;
+
+    const session = this.breakouts.get(sessionId);
+
+    if (!session) {
+      return;
+    }
+
+    session.parseRoster(locus);
+  },
+
+  /**
+   * Sets up listener for broadcast messages sent to the breakout session
+   */
+  listenToBroadcastMessages() {
+    this.listenTo(this.webex.internal.llm, 'event:breakout.message', (event) => {
```

Comment:
why not just `this.webex.internal.llm.on('event:breakout.message', ...)`

Reply:
This is again where taking advantage of the framework that ampersand state provides means you can worry less and write less code. `listenTo` is designed so that when you destroy the object you automatically stop listening to events. You don't have to worry that you've left the event handler there and need to clean it up manually.

Reply:
thank you for the explanation.